src/Collections/MemoryConfigCollection.php

```diff
@@ -10,6 +10,7 @@
 /**
  * Basic mutable config collection stored in memory
  */
+#[\AllowDynamicProperties]
```

Comment:
`__unserialize()` will only include dynamic properties _if they were already there when it was serialized_. So that's a red herring.
If we aren't actually using dynamic properties for this class anywhere outside the serialisation, then we should not add this attribute.
If we _are_ using dynamic properties for this class outside the serialisation, we should stop doing that.